api/v1alpha1/clusterinstance_validation.go

```diff
@@ -354,3 +809,21 @@ func validateReinstallGeneration(generation string) error {
 
 	return nil
 }
+
+// getReinstallPermissibleFields returns field names from ReinstallPermissions
+func getReinstallPermissibleFields() []string {
+	var fields []string
+	for field := range ReinstallPermissions {
+		fields = append(fields, field)
+	}
+	return fields
+}
+
+// getReinstallNodePaths returns JSON paths from ReinstallPermissions
+func getReinstallNodePaths() []string {
+	var paths []string
+	for _, path := range ReinstallPermissions {
+		paths = append(paths, path)
+	}
+	return paths
+}
```

Comment:
Minor nit, but I think these can just be `slices.Collect(maps.Keys(ReinstallPermissions))` and `slices.Collect(maps.Values(ReinstallPermissions))` in the caller (since there's only one), right?

Reply:
Indeed! Great suggestion 👍